influxql/parser.go

```diff
@@ -510,7 +510,7 @@ func (p *Parser) parseUInt64() (uint64, error) {
 // This function assumes the DURATION token has already been consumed.
 func (p *Parser) parseDuration() (time.Duration, error) {
 	tok, pos, lit := p.scanIgnoreWhitespace()
-	if tok != DURATION_VAL && tok != INF {
+	if tok != DURATION_VAL && tok != INF && tok != NUMBER {
```

Comment:
I removed the commit that modified parseDuration from this PR. Good eye, @otoolep. I was confused by the inconsistency with `FormatDuration` not printing units when the duration had to be represented in microseconds. That fix is in https://github.com/influxdb/influxdb/pull/4891.